src/ProtocolStreamReader.php

```diff
@@ -54,6 +54,9 @@ public function __construct($input)
                             $this->parsingMode = ParsingMode::HEADERS;
                             $this->headers = [];
                             $this->buffer = '';
+
+                            // after reading a full message, leave to allow different tasks to run
+                            return;
```

Comment:
We need some way to exit the while loop. (To serve setTimeouts, etc.) Without the return statement, the loop would only exit on EOF.